src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMStep.java

```diff
@@ -0,0 +1,244 @@
+        if (isBlank(projectName)) {
+            LOGGER.info("Error creating the Bitbucket SCM: The project name is blank");
+            projectKey = "";
+            repositorySlug = "";
+            repositoryUrl = "";
+            cloneUrl = "";
+            return;
+        }
+        if (isBlank(repositoryName)) {
+            LOGGER.info("Error creating the Bitbucket SCM: The repository name is blank");
+            projectKey = "";
+            repositorySlug = "";
+            repositoryUrl = "";
+            cloneUrl = "";
+            return;
+        }
+        BitbucketRepository repository = scmHelper.getRepository(projectName, repositoryName);
+        projectKey = repository.getProject().getKey();
+        repositorySlug = repository.getSlug();
+        cloneUrl = repository.getCloneUrls()
+                .stream()
+                .filter(link -> "http".equals(link.getName()))
+                .findFirst()
+                .map(BitbucketNamedLink::getHref)
+                .orElse("");
+        String selfLink = repository.getSelfLink();
+        repositoryUrl = selfLink.substring(0, max(selfLink.indexOf("/browse"), 0)); // self-link include /browse which needs to be trimmed
+    }
+
+    public List<BranchSpec> getBranches() {
+        return branches;
+    }
+
+    public String getCloneUrl() {
+        return cloneUrl;
+    }
+
+    public String getCredentialsId() {
+        return credentialsId;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getProjectKey() {
+        return projectKey;
+    }
+
+    public String getProjectName() {
+        return projectName;
+    }
+
+    public String getRepositoryName() {
+        return repositoryName;
+    }
+
+    public String getRepositorySlug() {
+        return repositorySlug;
+    }
+
+    public String getRepositoryUrl() {
+        return repositoryUrl;
+    }
+
+    public String getServerId() {
+        return serverId;
+    }
+
+    @Nonnull
+    @Override
+    protected SCM createSCM() {
+        return new BitbucketSCM(id, branches, cloneUrl, credentialsId, null, null, projectName, projectKey,
+                repositoryName, repositorySlug, repositoryUrl, serverId);
+    }
+
+    @Symbol("BitbucketSCMStep")
+    @Extension
+    public static final class DescriptorImpl extends SCMStepDescriptor implements BitbucketScmFormValidation, BitbucketScmFormFill {
+
+        @Inject
+        private BitbucketClientFactoryProvider bitbucketClientFactoryProvider;
+        @Inject
+        private BitbucketPluginConfiguration bitbucketPluginConfiguration;
+        @Inject
+        private BitbucketScmFormFillDelegate formFill;
+        @Inject
+        private BitbucketScmFormValidationDelegate formValidation;
+
+        @Override
+        @POST
+        public FormValidation doCheckCredentialsId(@QueryParameter String credentialsId) {
+            return formValidation.doCheckCredentialsId(credentialsId);
```

Comment:
This does not do any permission checks, it probably should.
The others do in the delegate but this done does not.